qubesadmin/app.py

```diff
@@ -472,6 +475,24 @@ def clone_vm(self, src_vm, new_name, new_cls=None, pool=None, pools=None,
             del self.domains[dst_vm.name]
             raise
 
+        if not ignore_devices:
+            try:
+                for devclass in src_vm.devices:
+                    for assignment in src_vm.devices[devclass].assignments(
+                            persistent=True):
+                        new_assignment = qubesadmin.devices.DeviceAssignment(
+                            backend_domain=assignment.backend_domain,
+                            ident=assignment.ident,
+                            options=assignment.options,
+                            persistent=assignment.persistent,
+                            frontend_domain=dst_vm,
+                            devclass=assignment.devclass)
```

Comment:
You don't need to set frontend_domain and devclass - those will be handled automatically. But it isn't a problem either (there is an assert in `attach()` that prevents invalid value).